packages/backend/src/open_payments/client/service.ts

```diff
@@ -68,3 +164,85 @@ export async function getIncomingPayment(
     return undefined
   }
 }
+
+async function getIncomingPayment(
+  deps: ServiceDependencies,
+  url: string
+): Promise<IncomingPaymentJSON | undefined> {
+  const requestHeaders = {
+    Authorization: `GNAP ${deps.accessToken}`,
+    'Content-Type': 'application/json'
+  }
+  try {
+    const { status, data } = await axios.get(url, {
+      headers: requestHeaders,
+      timeout: REQUEST_TIMEOUT,
+      validateStatus: (status) => status === 200
+    })
+    if (
+      !deps.validateIncomingPayment({
+        status,
+        body: data
+      }) ||
+      !isValidIncomingPayment(data)
+    ) {
+      throw new Error('unreachable')
+    }
+    return data
+  } catch (_) {
+    return undefined
+  }
+}
+
+const CONNECTION_URL_REGEX = /\/connections\/(.){36}$/
+const INCOMING_PAYMENT_URL_REGEX = /\/incoming-payments\/(.){36}$/
```

Comment:
The id doesn't necessarily have to be a uuid so I'd remove the `(.){36}` constraint. Maybe just check that it is alpha-numerical?

Reply:
It's currently enforced as `uuid` in the OpenAPI spec:
https://github.com/interledger/open-payments/blob/6d128e2298301a3a76d2421581e2ffa097ebf33c/open-api-spec.yaml#L629-L636
and this would make it even more so:
https://github.com/interledger/open-payments/pull/174/files#diff-aaff56ee0c5bb1e404fc4e010a7e38034fbc755542d28199e4cf0acb5ef580c9R625

Reply:
Interesting 🤔. I'm wondering whether we actually want to enforce that in the spec. It seems to me like more of a business decision.